Add warning of potential client/server incompatibilities (#17210)
PrefectHQ/prefect: docs/v3/manage/self-host.mdx

## Self-host a Prefect server

<Warning>

> Though old clients are compatible with new servers, we recommend using the same version of `prefect` for the client and server.

Reviewer: I suggest leading with the main recommendation.

```suggestion
We recommend using the same version of `prefect` for the client and server.
```

Reviewer 2: it seems important to include the information that old clients work with new servers, so imo whatever the order, it needs to be in there

Reviewer: Agreed. I accidentally left this as a single comment. See below.

> As new fields are added to the REST API, using a newer client version with an older server version can lead to incompatibilities or unexpected behavior. 

Reviewer: The second sentence can address what happens if you choose to ignore the best practice.

```suggestion
Old clients are compatible with new servers, but new clients can be incompatible with old servers as new fields are added to the REST API.
```

> These issues will normally manifest as responses from the server with a 422 status code.

Reviewer: ```suggestion
The server will typically return a 422 status code if any issues occur.
```
